### PR TITLE
for statement iterating through String characters

### PR DESCRIPTION
**Pull request by pkowal1982**
Would be useful to have it documented in for description

### CODE REVIEW

File: reference/gdscript.rst [hunk 1]
```diff
@@ -648,6 +648,8 @@ in the loop variable.
     for i in range(2,8,2):
         statement  # similar to [2, 4, 6] but does not allocate an array
 
+    for c in "Hello":
+        print(c)   # iterate through all characters in a String, print every letter in new line
```
Comment (bojidar-bg) [inline reference/gdscript.rst]: Isn't `on` better than `in` here?

**Comment by bojidar-bg** [comment]
There is still a merge conflict, can you try pulling master with --rebase?

**Comment by pkowal1982** [comment]
ok, conflict should be gone

File: reference/gdscript.rst [hunk 2]
```diff
@@ -696,6 +696,8 @@ in the loop variable.
     for i in range(2,8,2):
         statement  # similar to [2, 4, 6] but does not allocate an array
 
+    for c in "Hello":
+        print(c)   # iterate through all characters in a String, print every letter on new line
 match
```
Comment (akien-mga) [inline reference/gdscript.rst]: Would have been nice to keep a newline before the `match` section... Hope sphinx doesn't mess up the html rendering.

**Pull request merged**